lib/semantics/check-allocate.cc

```diff
@@ -37,10 +40,54 @@ struct AllocateCheckerInfo {
 class AllocationCheckerHelper {
 public:
   AllocationCheckerHelper(
-      const parser::AllocateObject &obj, AllocateCheckerInfo &info)
-    : allocateInfo_{info}, name_{parser::GetLastName(obj)},
+      const parser::Allocation &alloc, AllocateCheckerInfo &info)
+    : allocateInfo_{info}, allocateObject_{std::get<parser::AllocateObject>(
+                               alloc.t)},
+      name_{parser::GetLastName(allocateObject_)},
       type_{name_.symbol ? name_.symbol->GetType() : nullptr},
-      isSubobject_{std::holds_alternative<parser::StructureComponent>(obj.u)} {
+      allocateShapeSpecRank_{ShapeSpecRank(alloc)},
+      rank_{name_.symbol ? name_.symbol->Rank() : 0},
+      allocateCoarraySpecRank_{CoarraySpecRank(alloc)},
+      corank_{name_.symbol ? name_.symbol->Corank() : 0} {
+    GatherAllocationBasicInfo();
+  }
+
+  bool RunChecks(SemanticsContext &context);
+
+private:
+  AllocateCheckerInfo &allocateInfo_;
```

Comment:
It's a small point, but we generally put data members last in a class definition, after any private member functions.

Reply:
Fixed